tests/e2e/specs/out-of-game/profile.spec.js

```diff
@@ -0,0 +1,344 @@
+import { myUser, playerOne, playerTwo } from '../../fixtures/userFixtures';
+import dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc';
+import GameStatus from '../../../../utils/GameStatus.json';
+
+dayjs.extend(utc);
+
+describe('Profile Page', () => {
+  beforeEach(() => {
+    cy.wipeDatabase();
+    cy.visit('/');
+  });
+
+  describe('Profile page content', function() {
+    beforeEach(() => {
+      cy.signupPlayer(myUser);
+    });
+
+    it('Displays username', function() {
+      cy.vueRoute('/my-profile');
+      cy.contains('h1', 'My Profile').should('be.visible');
+      cy.get('[data-cy=username]').should('contain', myUser.username);
+    });
+
+    it('Shows Discord section when not connected', () => {
+      cy.vueRoute('/my-profile');
+      cy.contains('h2', 'Discord');
+      cy.get('[data-cy=not-connected]').should('be.visible');
+    });
+
+    it('Shows Discord connected state', function() {
+      cy.vueRoute('/my-profile');
+
+      cy.window().its('cuttle.authStore')
+        .then((authStore) => {
+          authStore.$patch({
+            identities: [
+              {
+                provider: 'discord',
+                username: 'TestDiscordUser#1234'
+              }
+            ]
+          });
+        });
+
+      cy.contains('h2', 'Discord').should('be.visible');
+      cy.get('[data-cy=discord-username]')
+        .should('be.visible')
+        .should('contain', 'TestDiscordUser#1234');
+      cy.get('[data-cy=not-connected]').should('not.exist');
+    });
+
+    it('Shows fallback when no games', function() {
+      cy.vueRoute('/my-profile');
+      cy.contains('h1', 'My Profile').should('be.visible');
+      cy.contains('No games found').should('be.visible');
+    });
+  });
+
+
+
+  describe('Profile Page - Scrolling and game list tests', () => {
+    beforeEach(() => {
+      cy.signupOpponent(myUser);
+      cy.signupOpponent(playerOne);
+    });
+
+    it('Loads more games on scroll', function() {
+      const games = Array.from({ length: 30 }).map((_, i) => ({
+        name: `Game ${i + 1}`,
+        status: GameStatus.FINISHED,
+        isRanked: true,
+        createdAt: dayjs.utc().subtract(i, 'day')
+          .toDate(),
+        p0: this[`${playerOne.username}Id`],
+        p1: this[`${myUser.username}Id`],
+        winner: this[`${myUser.username}Id`],
+      }));
+
+      cy.loadFinishedGameFixtures(games);
+      cy.loginPlayer(myUser);
+      cy.vueRoute('/my-profile');
+
+      cy.get('[data-test="game-list-item"]').should('have.length.at.least', 5);
```

Comment:
Maybe also reasonable to assert that it's below the page limit?
```suggestion
      cy.get('[data-test="game-list-item"]').should('have.length.below', 21);
```